x/btcstaking/migrations/v2/store.go

```diff
@@ -16,7 +17,21 @@ func MigrateStore(
 	ctx sdk.Context,
 	s storetypes.KVStore,
 	c codec.BinaryCodec,
-	modifyParams func(ctx context.Context, p *types.Params) error) error {
+	modifyParams func(ctx context.Context, p *types.Params) error,
+	indexTxHash func(ctx context.Context, txHash *chainhash.Hash),
+) error {
+	if err := migrateParams(ctx, s, c, modifyParams); err != nil {
+		return err
+	}
+
+	if err := indexAllowedMultiStakingTxs(ctx, indexTxHash); err != nil {
```

Comment:
I don't know if this should be a migration, in my mind would be something we would input in the upgrade handler, same way it was done at v1

Reply:
In general, if we did not have migration already I would agree with this i.e putting it in upgrade handler would be cleaner.

But as we are doing the migration either way, imo, doing it here nicely hides this whole thing from the rest of the app (functions can be private etc.)